### ISSUE TITLE
IPSec connection config does not allow configuration of advanced settings

### ISSUE THREAD

**Issue by GoumJer** [oracle/terraform-provider-oci#1929]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Creating an IPSec tunnel via Terraform does not allow the configuration of the advanced settings (diffie-Hellman, Encryption algoritms, time-out settings. 

### New or Affected Resource(s)

oci_core_ipsec_connection_tunnel_management

### Potential Terraform Configuration

```hcl
resource "oci_core_ipsec_connection_tunnel_management" "test_ipsec_connection_tunnel_management" {
  ipsec_id  = oci_core_ipsec.test_ip_sec_connection.id
  tunnel_id = data.oci_core_ipsec_connection_tunnels.test_ip_sec_connection_tunnels.ip_sec_connection_tunnels[0].id

  #Optional
  bgp_session_info {
    customer_bgp_asn      = "1587232876"
    customer_interface_ip = "10.0.0.16/31"
    oracle_interface_ip   = "10.0.0.17/31"
   }
   ## Feature request starts here
   # Advanced Settings
    ike_initiation     = "Initiator or responder"
    nat-t                  = "auto"
    dead_peer_detect  = "initiate or respond"
    dead_peer_timeout = 20
    # Phase 1
    encryption_algoritm_1       = "AES_256_CBC"
    authentication_algoritm_1 = "SHA2_256"
    diffie-hellman_1                 ="GROUP20"
    ike_lifetime                         = 28800
    # Phase 2
    encryption_algoritm_2       = "AES_256_CBC"
    authentication_algoritm_2 = "SHA2_256"
    ipsec_lifetime                     = 28800
    diffie-hellman_2                 ="GROUP20"
   #################

  display_name  = "MyIPSecConnection"
  routing       = "BGP"
  shared_secret = "sharedSecret"
  ike_version   = "V1"
}
```

### References

N/A

**Comment by iejones** [comment]
A bug fix to set phase 1 and 2 tunnel details will be released on November 15th, 2023

**Comment by gbatra89** [comment]
Is this has been released?

**Comment by iejones** [comment]
This fix was released with version [v5.21.0](https://github.com/oracle/terraform-provider-oci/releases/tag/v5.21.0). See phase_one_details and phase_one_details in the [documentation](https://registry.terraform.io/providers/oracle/oci/latest/docs/resources/core_ipsec_connection_tunnel_management) and the [example](https://github.com/oracle/terraform-provider-oci/blob/v5.21.0/examples/networking/ipsec_connections/ipsec_connection.tf#L127).
